src/main/java/org/commcare/xml/ActionParser.java

```diff
@@ -38,7 +38,19 @@ public Action parse() throws InvalidStructureException, IOException, XmlPullPars
         Vector<StackOperation> stackOps = new Vector<>();
 
         XPathExpression relevantExpr = parseRelevancyExpr();
-        boolean isAutoLaunching = "true".equals(parser.getAttributeValue(null, "auto_launch"));
+
+        String autoLaunch = parser.getAttributeValue(null, "auto_launch");
+
+        String derivedAutoLaunchExpression = deriveAutoLaunchExpression(autoLaunch);
+        XPathExpression autoLaunchExpr = null;
+
+        try {
+            autoLaunchExpr = XPathParseTool.parseXPath(derivedAutoLaunchExpression);
```

Comment:
should there not be a null check before this?

Reply:
thanks, corrected.